changes/process_changes.py

```diff
@@ -0,0 +1,124 @@
+# coding: utf-8
+
+from __future__ import absolute_import, division, print_function
+
+import argparse as A
+from funcy import memoize
+from glob import glob
+import os
+from os import path as P
+import re
+import yaml
+
+
+TN_RE = re.compile(r"[A-Z]\d\d\d-\d\d\d")
+
+
+def all_entries():
+    return [f for f in glob('*.yaml') if not f == 'entry_schema.yaml']
+
+
+types_to_header = {
+    'api-change': 'breaking changes',
+    'new-feature': 'new features'
+}
+
+
+def create_rst(change_type="api-change"):
+    """
+    Create an RST file from all the change entries for a given ``change_type``.
+    """
+    header_chunk = types_to_header[change_type]
+
+    def header(title, header_char):
+        """
+        Format a header
+        """
+        return "{}\n{}\n".format(title, header_char * len(title))
+
+    def field(name, value):
+        """
+        Format a field list entry
+        """
+        return ":{}:\n    {}".format(name, value)
+
+    def format_apis(apis):
+        """
+        Format the API field content. If absent, return "all".
+        """
+        if apis:
+            return ", ".join(apis)
+        else:
+            return "all"
+
+    entries = []
+
+    for fname in all_entries():
+        tn = fname.split(".")[0]
+        with open(fname) as f:
+            entry = yaml.safe_load(f)
+            entry['tn'] = tn
+            validate_entry(tn, entry)
+            if entry['type'] == change_type:
+                entries.append(entry)
+
+    entries = list(sorted(entries, key=lambda e: e['date'], reverse=True))
+
+    print(header("Libadalang API {}".format(header_chunk), "#"))
+
+    for entry in entries:
+        print(header(entry["title"], "="))
+        print(entry["description"])
+        print(field("tn", entry["tn"]))
+        print(field("date", entry["date"]))
+        print(field("apis", format_apis(entry.get("apis"))))
+        print()
+
+
+@memoize
+def json_schema():
+    with open("entry_schema.yaml") as f:
```

Comment:
Looks like you forgot to add this schema file: can you add it?

Reply:
Good catch, thanks!